compiler/rustc_error_messages/locales/en-US/symbol_mangling.ftl

```diff
@@ -0,0 +1,7 @@
+symbol_mangling_invalid_symbol_name = symbol-name({$mangled_formatted})
+
+symbol_mangling_invalid_trait_item = demangling({$demangling_formatted})
+
+symbol_mangling_alt_invalid_trait_item = demangling-alt({$alt_demangling_formatted})
+
+symbol_mangling_invalid_def_path = def-path({$def_path})
```

Comment:
These can actually be all merged together, since they're `{$kind}({$contents})`.
Might even make more sense to use `{$kind} = {$contents}` or some other style, but that's a bit orthogonal.

Reply:
I think this might be a good way to make this go through the translation infrastructure - so that we can eventually remove the non-translation codepaths - while making translation of the message a no-op.

@JhonnyBillM do you think you could submit a new PR that makes this change? It's just that these "diagnostics" really only exist to be checked our the test suite to test the mangling scheme, and so names like they have (which I should have caught in review), might be misleading.

Reply:
Thank you for the review! Helped me understand better this crate and file.
Addressed your points in https://github.com/rust-lang/rust/pull/101782